lua/lint/linters/phpstan.lua

```diff
@@ -0,0 +1,33 @@
+return {
+  cmd = 'phpstan',
+  args = {
+    'analyze',
+    '--error-format=json',
+    '--no-progress',
+  },
+  ignore_exitcode = true,
+  parser = function(output)
+    if output == nil then
+      return {}
+    end
+
+    local file = vim.json.decode(output).files[vim.api.nvim_buf_get_name(0)]
```

Comment:
```suggestion
  parser = function(output, bufnr)
    if output == nil then
      return {}
    end
    local file = vim.json.decode(output).files[vim.api.nvim_buf_get_name(bufnr)]
```

The linting process runs asynchronous and by the time the parser function is called a user could've switched to another buffer.

Reply:
Didn't know about the `bufnr` argument, but makes sense. Already updated.